compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -668,10 +668,15 @@ where
         ct: AbstractConst<'tcx>,
         f: &mut dyn FnMut(AbstractConst<'tcx>) -> ControlFlow<R>,
     ) -> ControlFlow<R> {
-        f(ct)?;
         let root = ct.root(tcx);
-        debug!(?root);
-        match root {
+        if let Node::Leaf(leaf) = root {
+            if let Ok(Some(ac)) = AbstractConst::from_const(tcx, leaf) {
+                return recurse(tcx, ac, f);
+            }
+            return ControlFlow::CONTINUE;
```

Comment:
why are you returning here?

Reply:
For which one? I did `return recurse(...)` to omit the `Unevaluated`

Reply:
well, the goal is 
```rust

Reply:
Ah, I thought that by calling recurse it would implicitly capture the rest of the values, but I guess I'm confused about which one would actually be the branch hit when it is `Unevaluated(Non-Leaf)`

Reply:
the issue isn't calling recurse, the issue is the return if there is no nested `AbstractConst`